.github/container/Dockerfile.t5x.arm64

```diff
@@ -0,0 +1,226 @@
+        apt-utils \
+        build-essential \
+        checkinstall \
+        libffi-dev
+
+# Install pip dependencies needed for array_record
+RUN --mount=type=cache,target=/root/.cache \
+    ${PYTHON_BIN_PATH} -m pip install -U pip && \
+    ${PYTHON_BIN_PATH} -m pip install -U \
+        absl-py \
+        auditwheel \
+        etils[epath] \
+        patchelf \
+        setuptools \
+        twine \
+        wheel;
+
+RUN <<"EOT" bash -exu
+set -o pipefail
+
+git clone https://github.com/google/array_record.git /tmp/array_record
+cd /tmp/array_record
+git checkout v0.5.0
+
+export CROSSTOOL_TOP="@ml2014_aarch64_config_aarch64//crosstool:toolchain"
+export AUDITWHEEL_PLATFORM="manylinux2014_aarch64"
+./oss/build_whl.sh
+EOT
+
+
+#------------------------------------------------------------------------------
+# build grain from a manually-patched version
+#------------------------------------------------------------------------------
+# TODO: Remove this once grain maintainers have published an arm64 wheel.
+
+FROM wheel-builder as grain-builder
+ARG PYTHON_VERSION="3.10"
+
+RUN <<"EOT" bash -exu
+set -o pipefail
+
+# Setup python
+apt-get update && apt-get install -y \
+    python3-dev python3-pip python3-venv && \
+    rm -rf /var/lib/apt/lists/* && \
+    python${PYTHON_VERSION} -m pip install pip --upgrade && \
+    update-alternatives --install /usr/bin/python python /usr/bin/python${PYTHON_VERSION} 0
+
+# NOTE: We are ignoring the fact that "array_record" will actually resolve to the wrong version
+#   here. When we put it all together in the final image, we make sure that the right version is
+#   already installed.
+pip install \
+    absl-py \
+    array_record \
+    build \
+    cloudpickle \
+    dm-tree \
+    etils[epath] \
+    "more-itertools>=9.1.0" \
+    numpy;
+
+pip install \
+    dill \
+    jax \
+    jaxlib \
+    tensorflow \
+    tensorflow-datasets;
+
+git clone https://github.com/gspschmid/grain /opt/grain
+cd /opt/grain
+git checkout external-arm64-build-poc
+
+chmod a+x ./grain/oss/build_whl.sh
+./grain/oss/build_whl.sh
+
+ls /tmp/grain/all_dist/*.whl
+EOT
+
+
+###############################################################################
+## T5X for AArch64
+###############################################################################
+
+ARG BASE_IMAGE
+FROM ${BASE_IMAGE} as mealkit
+ARG REPO_T5X
+ARG REF_T5X
+ARG SRC_PATH_T5X
+
+COPY --from=array_record-builder /tmp/array_record/all_dist/array_record*linux_aarch64.whl /opt/
+RUN echo "array_record @ file://$(ls /opt/array_record*.whl)" >> /opt/pip-tools.d/manifest.pax
+
+COPY --from=grain-builder /tmp/grain/all_dist/grain*.whl /opt/
+RUN echo "grain @ file://$(ls /opt/grain*.whl)" >> /opt/pip-tools.d/manifest.pax
+
+
+### --- TO BE REFACTORED? (everything below is as in @nouiz's original PR) ---
+
+# force a recent version to have latest protobuf dep
+# Install now as some build need them.
+RUN pip install "tensorflow_datasets==4.9.2"
```

Comment:
In the production Dockerfile, we should not pip install packages in separate steps.
The purpose of these installs seem to provide the dependencies for building tftext. Could you please port the tftext builder section from the arm64 paxml Dockerfile to this PR?